.github/workflows/CD.yml

```diff
@@ -318,44 +323,31 @@ jobs:
         uses: actions/cache@v2
         with:
           key: "${{ env.PACKAGE_CACHE_KEY }}"
-          path: build/packages
+          path: $PACKAGING_ROOT
 
-      - name: Configure AWS Credentials
-        uses: aws-actions/configure-aws-credentials@v1
+      - name: Login to Public Release ECR
+        id: login-ecr
+        uses: docker/login-action@v1
         with:
-          aws-access-key-id: ${{ secrets.RELEASE_KEY_ID }}
-          aws-secret-access-key: ${{ secrets.RELEASE_SECRET }}
-          aws-region: us-west-2    
-          
-      - name: Install AWS Cli 2.0
-        run: |
-          curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2.zip"
-          unzip awscliv2.zip
-          sudo ./aws/install --update
+          registry: public.ecr.aws
+          username: ${{ secrets.RELEASE_KEY_ID }}
+          password: ${{ secrets.RELEASE_SECRET }}
+        env:
+          AWS_REGION: us-east-1
 
-      - name: Load Image
-        run: docker load < build/packages/$IMAGE_NAME.tar
-        
-      - name: upload to ECR
+      - name: Get version tag and testing version tag
+        id: versioning
         run: |
-          TAG=`cat build/packages/VERSION`
-          aws ecr-public get-login-password --region us-east-1 | docker login --username AWS --password-stdin public.ecr.aws/aws-observability
-          docker tag $IMAGE_NAME public.ecr.aws/$ECR_REPO:$TAG
-          docker push public.ecr.aws/$ECR_REPO:$TAG
+          Tag=`cat $PACKAGING_ROOT/VERSION`
```

Comment:
Instead of getting the version here, I would suggest outputting the version as part of https://github.com/aws-observability/aws-otel-collector/blob/85c027bf2347be126a77a48e308f9cfc64765b83/.github/workflows/CD.yml#L57-L68 and then adding the `release-checking` job to the `needs` of each job that needs the version. Would still name them `version` and `testing-version` instead of `tag` and `testing_tag` too in case they could be used elsewhere.

Reply:
Reasonable. Thought everything would make sense to keep it nearest to the building version. However, after considering the visibility and short-condition, make sense to me now.